plugin/advisor/catalog/walk_through.go

```diff
@@ -0,0 +1,420 @@
+	if _, exists := d.schemaSet[""]; !exists {
+		d.createSchema("")
+	}
+
+	nodeList, err := d.parse(stmts)
+	if err != nil {
+		return err
+	}
+
+	for _, node := range nodeList {
+		if err := d.changeState(node); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (d *databaseState) changeState(in tidbast.StmtNode) error {
+	switch node := in.(type) {
+	case *tidbast.CreateTableStmt:
+		return d.createTable(node)
+	default:
+		return nil
+	}
+}
+
+func (d *databaseState) createTable(node *tidbast.CreateTableStmt) error {
+	if node.Table.Schema.O != "" && d.name != node.Table.Schema.O {
+		return &WalkThroughError{
+			Type:    ErrorTypeAccessOtherDatabase,
+			Content: fmt.Sprintf("Database `%s` is not the current database", node.Table.Schema.O),
+		}
+	}
+
+	schema, exists := d.schemaSet[""]
+	if !exists {
+		schema = d.createSchema("")
+	}
+
+	if _, exists = schema.tableSet[node.Table.Name.O]; exists {
+		if node.IfNotExists {
+			return nil
+		}
+		return &WalkThroughError{
+			Type:    ErrorTypeTableExists,
+			Content: fmt.Sprintf("Table `%s` exists", node.Table.Name.O),
+		}
+	}
+
+	table := &tableState{
+		name:      node.Table.Name.O,
+		columnSet: make(columnStateMap),
+		indexSet:  make(indexStateMap),
+	}
+	schema.tableSet[table.name] = table
+
+	for i, column := range node.Cols {
+		if err := table.createColumn(column, i+1); err != nil {
+			return err
+		}
+	}
+
+	for _, constraint := range node.Constraints {
+		if err := table.createConstraint(constraint); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (t *tableState) createConstraint(constraint *tidbast.Constraint) error {
+	switch constraint.Tp {
+	case tidbast.ConstraintPrimaryKey:
+		keyList, err := t.validateAndGetKeyStringList(constraint.Keys, true)
+		if err != nil {
+			return err
+		}
+		if err := t.createPrimaryKey(keyList, getIndexType(constraint.Option)); err != nil {
+			return err
+		}
+	case tidbast.ConstraintKey, tidbast.ConstraintIndex:
+		keyList, err := t.validateAndGetKeyStringList(constraint.Keys, false)
+		if err != nil {
+			return err
+		}
+		if err := t.createIndex(constraint.Name, keyList, false, getIndexType(constraint.Option)); err != nil {
+			return err
+		}
+	case tidbast.ConstraintUniq, tidbast.ConstraintUniqKey, tidbast.ConstraintUniqIndex:
+		keyList, err := t.validateAndGetKeyStringList(constraint.Keys, false)
+		if err != nil {
+			return err
+		}
+		if err := t.createIndex(constraint.Name, keyList, true, getIndexType(constraint.Option)); err != nil {
+			return err
+		}
+	case tidbast.ConstraintForeignKey:
+		// we can not deal with FOREIGN KEY constraints
+	case tidbast.ConstraintFulltext:
+		keyList, err := t.validateAndGetKeyStringList(constraint.Keys, false)
+		if err != nil {
+			return err
+		}
+		if err := t.createIndex(constraint.Name, keyList, false, FullTextName); err != nil {
+			return err
+		}
+	case tidbast.ConstraintCheck:
+		// we can not deal with CHECK constraints
```

Comment:
Is this a TODO and should return error?

Reply:
The reason is the data model in Bytebase. We don't collect the CONSTRAINT information, so I think we should skip these statements instead of returning errors.

Reply:
In other words, we are not interested in these statements.

Reply:
Then this is "Do not" instead of "Can not" and would better elaborate a bit reason here that Bytebase doesn't collect such info

Reply:
Aha, a nice advice!